appcenter-auth/android/src/main/java/com/microsoft/appcenter/reactnative/auth/AppCenterReactNativeAuthModule.java

```diff
@@ -27,4 +39,54 @@ public AppCenterReactNativeAuthModule(Application application) {
     public String getName() {
         return "AppCenterReactNativeAuth";
     }
+
+    @ReactMethod
+    public void isEnabled(final Promise promise) {
+        Auth.isEnabled().thenAccept(new AppCenterConsumer<Boolean>() {
+
+            @Override
+            public void accept(Boolean isEnabled) {
+                promise.resolve(isEnabled);
+            }
+        });
+    }
+
+    @ReactMethod
+    public void setEnabled(boolean enabled, final Promise promise) {
+        Auth.setEnabled(enabled).thenAccept(new AppCenterConsumer<Void>() {
+
+            @Override
+            public void accept(Void aVoid) {
+                promise.resolve(null);
+            }
+        });
+    }
+
+    @ReactMethod
+    public void signIn(final Promise promise) {
+        Auth.signIn().thenAccept(new AppCenterConsumer<SignInResult>() {
+
+            @Override
+            public void accept(SignInResult signInResult) {
+                WritableMap writableMap = new WritableNativeMap();
+                if (signInResult.getException() == null) {
+
+                    /* Sign-in succeeded, convert Java result to a JavaScript result. */
+                    UserInformation userInformation = signInResult.getUserInformation();
+                    writableMap.putString(ACCESS_TOKEN_KEY, userInformation.getAccessToken());
+                    writableMap.putString(ACCOUNT_ID_KEY, userInformation.getAccountId());
+                    writableMap.putString(ID_TOKEN_KEY, userInformation.getIdToken());
+                    promise.resolve(writableMap);
+                } else {
+                    Exception signInFailureException = signInResult.getException();
+                    promise.reject("sign_in_failed", "Sign-in failed", signInFailureException);
+                }
+            }
+        });
+    }
+
+    @ReactMethod
+    public void signOut() {
+        Auth.signOut();
```

Comment:
Resolve the promise with `null` on void method.

```suggestion
        Auth.signOut();
        promise.resolve(null);
```

Reply:
We don't use promise for synchronous APIs on several Java APIs.
It's not consistent though.
For simplicity I would keep that one without the promise like we did when binding the AppCenter core in Java.

Reply:
Agreed with @guperrot, we don't need it.